Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,572 @@
+
+    // get a reference to WarpX instance
+    auto& warpx = WarpX::GetInstance();
+
+    // get cell size
+    amrex::Geometry const & geom = warpx.Geom(0);
+#if defined(WARPX_DIM_1D_Z)
+        amrex::Real dV = geom.CellSize(0);
+#elif defined(WARPX_DIM_XZ)
+        amrex::Real dV = geom.CellSize(0) * geom.CellSize(1);
+#elif defined(WARPX_DIM_3D)
+        amrex::Real dV = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2);
+#endif
+
+    const auto get_idx = [&](const std::string& name){
+        return m_headers_indices.at(name).idx;
+    };
+
+    std::array<std::unique_ptr<amrex::MultiFab>, 2> num_dens;
+
+    // loop over species
+    for (int i_s = 0; i_s < 2; ++i_s)
+    {
+        // get WarpXParticleContainer class object
+        WarpXParticleContainer& myspc = mypc.GetParticleContainerFromName(m_beam_name[i_s]);
+        // get charge
+        amrex::ParticleReal const q = myspc.getCharge();
+
+        using PType = typename WarpXParticleContainer::SuperParticleType;
+
+        num_dens[i_s] = myspc.GetChargeDensity(0);
+        num_dens[i_s]->mult(1./q);
+
+#if defined(WARPX_DIM_XZ)
+        // w_tot
+        // x_ave,
+        // thetax_min, thetax_ave, thetax_max
+        amrex::ReduceOps<ReduceOpSum,
+                         ReduceOpSum,
+                         ReduceOpMin, ReduceOpSum, ReduceOpMax> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<Real,
+                                                         Real,
+                                                         Real, Real, Real>>(
+            myspc,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<Real,
+                                                                             Real,
+                                                                             Real, Real, Real>
+            {
+                const amrex::Real w  = p.rdata(PIdx::w);
+                const amrex::Real x = p.pos(0);
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                return {w, w*x,
+                        thetax, w*thetax, thetax};
+            },
+            reduce_ops);
+
+        amrex::Real w_tot  = amrex::get<0>(r);
+        amrex::Real x_ave  = amrex::get<1>(r);
+        amrex::Real thetax_min  = amrex::get<2>(r);
+        amrex::Real thetax_ave  = amrex::get<3>(r);
+        amrex::Real thetax_max  = amrex::get<4>(r);
+
+        amrex::ParallelDescriptor::ReduceRealSum({w_tot, x_ave, thetax_ave},
+                                                  amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMin(thetax_min, amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMax(thetax_max, amrex::ParallelDescriptor::IOProcessorNumber());
+
+        if (w_tot > std::numeric_limits<Real>::min())
```

Comment:
Is there a reason that this isn't the more simple `w_tot > 0.`?

Reply:
Not that I know of. I will simplify it.